include/ofi_xpmem.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * (C) Copyright 2023 UT-Battelle, LLC. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef OFI_XPMEM_H
+#define OFI_XPMEM_H
+
+#include <ofi_mr.h>
+#include <ofi_util.h>
+
+#if HAVE_XPMEM
+#include <xpmem.h>
+#else
+#define XPMEM_RDWR -1
+#define XPMEM_PERMIT_MODE -1
+#define xpmem_get(seg_id, flags, perm_type, perm_value) -1
+#define xpmem_release(apid)
+typedef int64_t xpmem_apid_t;
+typedef int64_t xpmem_segid_t;
+#endif
```

Comment:
Can you add a comment to the end of this end if to show which if it goes with, please? And maybe also a comment to explain that this is needed for the smr_region to compile without xpmem available? I stared at this for a little too long lol